### PR TITLE
Adds tests cases for make_* system macros

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

#88 

**Description of changes:**

Adds actual test cases for all of the `make_*` system macros.

Things to review:
* Do the test cases accurately implement the descriptions?
* Are there any edge cases that are missing?
* Are the parameter encoding tests for `make_timestamp` correct? (I think they are, but I need confirmation.)

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: conformance/system_macros/make_timestamp.ion [hunk 1]
```diff
@@ -1,22 +1,305 @@
...
+         (each "in text with an unqualified macro name"
+               (text " (:make_timestamp 1) ")
+               "in text with an unqualified macro address"
+               (text " (:7 1) ")
+               "in text with a qualified macro name"
+               (text " (:$ion::make_timestamp 1) ")
+               "in text using qualified system macro address 7"
+               (text " (:$ion::7 1) ")
+               "in binary using system macro address 7"
+               (binary "EF 07 00 00 01 00")
```
Comment (zslayton) [inline conformance/system_macros/make_timestamp.ion:14]: Interesting, `make_timestamp` is the only system macro that requires a 2-byte arg encoding bitmap. I think this is probably ok since most of its value proposition comes from usage in TDL, but it is eye-catching.

**Review by zslayton — APPROVED** [review]
🚀

File: conformance/system_macros/make_field.ion [hunk 2]
```diff
@@ -1,10 +1,68 @@
...
+                     "a symbol with known text"
+                     (text "(:make_field foo 0)")
+                     "an expression that produces text"
+                     (text "(:make_field (:values 'foo') 0)")
+                     "annotated"
+                     (text "(:make_field bar::foo 0)")
+                     (produces {foo:0} ))
+               (then "a symbol with unknown text"
+                     (text "(:make_field $0 1)")
+                     // Could be (produces {$0:1} ), but some implementations don't support $0 nicely.
```
[the remark this answers is not in the document]
Comment (popematt) [reply conformance/system_macros/make_field.ion:32]: Yeah, and it's also exposing a weakness in `IonElement`—it doesn't have any support for `$0`, but we're using it in the conformance runner in `ion-java` because it's so much easier to work with than `IonValue`.

**Review by tgregg — APPROVED** [review]
